services/udp/service.go

```diff
@@ -153,15 +156,17 @@ func (s *Service) serve() {
 			return
 		default:
 			// Keep processing.
-			buf := make([]byte, s.config.UDPPayloadSize)
 			n, _, err := s.conn.ReadFromUDP(buf)
 			if err != nil {
 				s.statMap.Add(statReadFail, 1)
 				s.Logger.Printf("Failed to read UDP message: %s", err)
 				continue
 			}
 			s.statMap.Add(statBytesReceived, int64(n))
-			s.parserChan <- buf[:n]
+
+			bufCopy := make([]byte, n)
+			copy(bufCopy, buf[:n])
+			s.parserChan <- bufCopy
```

Comment:
I'm not sure why I thought (or remembered) differently, but as long as this checks out and passes tests then I'm fine with it.